tests/test_training.py

```diff
@@ -52,30 +86,40 @@ def gpt_tiny_cfg(conf_path: str = 'scripts/train/yamls/pretrain/mpt-125m.yaml'):
                      not torch.cuda.is_available(),
                      reason='testing with cuda requires GPU')),
 ])
-@pytest.mark.parametrize('logit_scale', [None, 0.036, 'inv_sqrt_d_model'])
-def test_train(device: str, logit_scale: Optional[Union[float, str]]):
-    if not os.path.isdir('./my-copy-c4/val'):
-        pytest.xfail('c4 dataset not set up as expected')
-
-    warnings.filterwarnings(
-        action='ignore',
-        category=DeprecationWarning,
-        message=
-        "Using the 'grad_clip_norm' field in Trainer is deprecated. Please usethe GradientClipping Algorithm in composer.algorithms.gradient_clipping."
-    )
-
-    test_cfg = gpt_tiny_cfg(
-        conf_path='scripts/train/yamls/pretrain/mpt-125m.yaml')
-    test_cfg.eval_subset_num_batches = 2
-    if logit_scale:
-        test_cfg.model.logit_scale = logit_scale
+def test_train(device: str):
+    create_c4_dataset_xsmall()
+
+    conf_path: str = os.path.join(repo_dir,
+                                  'scripts/train/yamls/pretrain/mpt-125m.yaml')
+    with open(conf_path) as f:
+        test_cfg = om.load(f)
+
+    assert isinstance(test_cfg, DictConfig)
+    test_cfg.data_local = 'my-copy-c4'
+    test_cfg.max_duration = '1ba'
+    test_cfg.eval_interval = 0
+    test_cfg.save_folder = 'mpt-125m'
+    test_cfg.save_overwrite = True
+
+    test_cfg.model.d_model = 32
+    test_cfg.model.n_heads = 2
+    test_cfg.model.n_layers = 2
+    test_cfg.max_seq_len = 256
+    test_cfg.model.max_seq_len = test_cfg.max_seq_len
+    test_cfg.tokenizer.kwargs.model_max_length = test_cfg.max_seq_len
+    test_cfg.train_loader.dataset.max_seq_len = test_cfg.max_seq_len
+    test_cfg.eval_loader.dataset.max_seq_len = test_cfg.max_seq_len
 
     if device == 'cpu':
-        pytest.xfail(
-            'FSDP in PyTorch 1.13 does not support precision `Precision.FP32` with sharding_strategy `FULL_SHARD.`'
-        )
         test_cfg.model.init_device = 'cpu'
-        test_cfg.model.attn_impl = 'torch'
+        test_cfg.fsdp_config = None
+        test_cfg.model.attn_config.attn_impl = 'torch'
+        test_cfg.model.loss_fn = 'torch_crossentropy'
         test_cfg.precision = 'fp32'
 
-    main(test_cfg)
+    trainer = main(test_cfg)
+    trainer_loss: float = trainer.state.loss.item()  # type: ignore
+
+    expected_golden_loss = 11.80
```

Comment:
non-blocking comment

The magic number test seems like a recipe for future failures...

Reply:
Yeah let's see how it goes. Easy enough to fix if it is flaky

Reply:
agreed for flakiness; let's remove it, I'll just add an assert True at the end